app/adapter/db/allocatedkey_transactional.go

```diff
@@ -48,9 +40,11 @@ VALUES ($1, $2);
 			return err
 		}
 	}
-	return tx.Commit()
+
+	return nil
 }
 
-func NewAllocatedKeySQL(db *sql.DB) AllocatedKeySQL {
-	return AllocatedKeySQL{db: db}
+// NewAllocatedKeyTransactional creates a new instance of AllocatedKeyTransactional
+func NewAllocatedKeyTransactional(tx *sql.Tx) AllocatedKeyTransactional {
```

Comment:
```suggestion
func NewAllocatedKeyTransactional(db *sql.DB) AllocatedKeyTransactional {
```

Sorry that I didn't make it clear. I actually mean passing `*sql.DB` into the constructor. SQL transaction is implementation detail for each operation. To me, we shouldn't aggregate all calls to `CreateInBatch` as single transaction at SQL level.